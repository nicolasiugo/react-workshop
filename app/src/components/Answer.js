import React from 'react'
import PropTypes from 'prop-types';


const Answer = (props) => {
  const {
    answerId,
    onAnswerSelected,
    answerContent
  } = props;

  return (
    <div>
      <input 
        id={`answer-input-${answerId}`}
        value={answerId}
        defaultChecked={false}
        type="radio"
        onChange={onAnswerSelected}/>
      <label htmlFor={`answer-input-${answerId}`}>
        {answerContent}
      </label>
    </div>
  );
}

Answer.propTypes = {
  answerId: PropTypes.number.isRequired,
  answerContent: PropTypes.string.isRequired,
  onAnswerSelected: PropTypes.func.isRequired
}

export default Answer;