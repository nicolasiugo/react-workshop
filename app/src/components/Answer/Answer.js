import React from 'react'
import PropTypes from 'prop-types'


const Answer = (props) => {
  const {
    answerId,
    onAnswerSelected,
    answerContent,
    isChecked
  } = props;

  return (
    <div>
      <input 
        id={`answer-input-${answerId}`}
        value={answerId}
        checked={isChecked}
        type="radio"
        onChange={onAnswerSelected}/>
      <label htmlFor={`answer-input-${answerId}`}>
        {answerContent}
      </label>
    </div>
  );
}

Answer.defaultProps = {
  isChecked: false
}

Answer.propTypes = {
  answerId: PropTypes.number.isRequired,
  isChecked: PropTypes.bool,
  answerContent: PropTypes.string.isRequired,
  onAnswerSelected: PropTypes.func.isRequired
}

export default Answer;