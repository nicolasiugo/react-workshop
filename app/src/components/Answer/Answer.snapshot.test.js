import React from 'react';
import Answer from './Answer';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const mockFn = jest.fn();
  const tree = renderer
    .create(
      <Answer 
        answerId={1}
        answerContent="Una respuesta"
        onAnswerSelected={mockFn}/>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
