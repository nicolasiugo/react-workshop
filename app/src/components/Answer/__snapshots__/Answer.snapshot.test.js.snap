exports[`test renders correctly 1`] = `
<div>
  <input
    checked={false}
    id="answer-input-1"
    onChange={[Function]}
    type="radio"
    value={1} />
  <label
    htmlFor="answer-input-1">
    Una respuesta
  </label>
</div>
`;
